packages/api/src/command/medical/cq-directory/search-cq-directory.ts

```diff
@@ -35,16 +43,26 @@ export async function searchNearbyCQOrganizations({
   radiusInMiles?: number;
 }): Promise<CQOrgBasicDetails[]> {
   const radiusInMeters = convert(radiusInMiles).from("mi").to("m");
-
   const patient = await getPatientOrFail({ id: patientId, cxId });
-  const coordinates = await geocodeAddresses(patient.data.address);
+  const addresses = patient.data.address;
+  const needUpdate = await addGeographicCoordinates(addresses);
+  if (needUpdate) {
+    const updData = patientUpdateSchema.parse(patient.data);
+    const patientUpdate: PatientUpdateCmd = {
+      ...schemaUpdateToPatient(updData, cxId),
+      id: patient.id,
+    };
+    await updatePatient(patientUpdate);
+  }
+  const coordinates = addresses.flatMap(address => address.coordinates ?? []);
+  if (!coordinates.length) throw new Error("Failed to get patient coordinates");
```

Comment:
Also, not sure if worth, but maybe I should modularize this more, and have the logic for updating the patient on `patient.ts`, and have it call `addGeographicCoordinates` which deals with a single address instead of an array